source/MissionAction.h

```diff
@@ -40,6 +40,22 @@ class UI;
 // include showing a dialog or conversation, giving the player payment or a
 // special item, modifying condition flags, or queueing an event to occur.
 class MissionAction {
+public:
+	
+	class CheckLocations {
+	public:
+		bool empty = true;
+		
+		bool flag = false;
```

Comment:
I think it could make more sense to use a bit flag like the NPC `actions` and an enum for these values. If you prefer having them "separate" still, you could use a `vector<bool>` with the enums being associated with valid vector indices.